### DISCUSSION TITLE
useTransition smooth text animation

### DISCUSSION THREAD

**Discussion by victors1681** [pmndrs/react-spring#776] [Support]
## 🤓 Question

I'm trying to animate sentences using transition. 
1-Get the text `Lorem Ipsum is simply dummy text of the printing.`
2-Separating the words `const arr = text.split(' ')`
3-Animate each item. 

This approach is working but went I pass another text I have a weird transition. 
when I set the new state with the new text start appearing before the previous text is gone and the new text starts to relocate.

![bad-animation](https://user-images.githubusercontent.com/11906852/62817042-3844c000-bafe-11e9-8ef4-951498b57e9e.gif)
**Desire behaviors** 
Smooth transition at the beginning, when a new text comes, old text disappear smoothly and the new one appears in the same position as the previous one.

Sample code:
https://codesandbox.io/s/frosty-darkness-66izj?fontsize=14

I'll appreciate your help guys.

**Comment by clemente-xyz** [comment]
Hi guys! Any updates on this issue?

**Comment by noeljackson** [comment]
Have you tried letting useTransition set the key instead of your UUID function?
